### PR TITLE
CSP-367 - PATCH Admin

### PR DESCRIPTION
**Pull request by LiamWright**
PATCH /Admin endpoint

### CODE REVIEW

File: src/SFA.DAS.AANHub.Application.UnitTests/Admins/Commands/PatchAdminMember/PatchAdminMemberCommandHandlerTests.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+                RequestedByMemberId = requestedByMemberId,
+                PatchDoc = patchDoc
+            };
+
+            var response = await sut.Handle(command, cancellationToken);
+            Assert.NotNull(response);
+            response.Result.IsSuccess.Should().BeTrue();
+            response.IsValidResponse.Should().BeTrue();
+            response.Errors.Count.Should().Be(0);
+
```
Comment (DevenKShah) [inline src/SFA.DAS.AANHub.Application.UnitTests/Admins/Commands/PatchAdminMember/PatchAdminMemberCommandHandlerTests.cs:44]: Asserting on updated date would help. You could manually change the date right at the top to AddDay(-1) or something  and then assert later by checking the entity that was fed into write repository to have different date. 
Or even better, inject a ITimeProvider as explained on slack.

**Review by DevenKShah — APPROVED** [review]
Approving  but a little extra coverage is required around LastUpdated date.

File: src/SFA.DAS.AANHub.Application.UnitTests/Admins/Commands/PatchAdminMember/PatchAdminMemberCommandHandlerTests.cs [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+            Assert.AreEqual(emailValue, admin.Email);
+            Assert.AreEqual(nameValue, admin.Name);
+
+            Assert.NotNull(admin.LastUpdated);
+
+            auditWriteRepository.Verify(a => a.Create(It.Is<Audit>(x =>
+                getAudit(x!.After!).LastUpdated.GetValueOrDefault().Day == admin.LastUpdated.GetValueOrDefault().Day)));
+
+            ;
```
[the remark this answers is not in the document]
Comment (LiamWright) [reply src/SFA.DAS.AANHub.Application.UnitTests/Admins/Commands/PatchAdminMember/PatchAdminMemberCommandHandlerTests.cs]: Have deleted, strange that the IDE didn't pick this up!

File: src/SFA.DAS.AANHub.Application.UnitTests/Admins/Commands/PatchAdminMember/PatchAdminMemberCommandHandlerTests.cs [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+using SFA.DAS.AANHub.Domain.Interfaces.Repositories;
+using static SFA.DAS.AANHub.Domain.Common.Constants;
+
+namespace SFA.DAS.AANHub.Application.UnitTests.Admins.Commands.PatchAdminMember
+{
+    public class PatchAdminMemberCommandHandlerTests
+    {
+        private const string Email = "Email";
+        private const string Name = "Name";
+        private const string Organisation = "Organisation";
```
[the remark this answers is not in the document]
Comment (LiamWright) [reply src/SFA.DAS.AANHub.Application.UnitTests/Admins/Commands/PatchAdminMember/PatchAdminMemberCommandHandlerTests.cs]: Removed

File: src/SFA.DAS.AANHub.Api/Controllers/AdminsController.cs [hunk 4]
```diff
@@ -73,5 +76,37 @@ public async Task<IActionResult> GetAdmin(string userName)
...
+        /// <param name="userName"></param>
+        /// <param name="request"></param>
+        /// ///
+        [HttpPatch]
+        [Route("{userName}")]
+        [Produces("application/json")]
+        [ProducesResponseType(typeof(string), StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(typeof(NoContentResult), StatusCodes.Status204NoContent)]
+        [ProducesResponseType(typeof(NotFoundResult), StatusCodes.Status404NotFound)]
+        public async Task<IActionResult> PatchAdmin([FromHeader(Name = Constants.PostRequestHeaders.RequestedByUserHeader)] [Required] Guid userId,
```
[the remark this answers is not in the document]
Comment (LiamWright) [reply src/SFA.DAS.AANHub.Api/Controllers/AdminsController.cs]: Updated name in controller and tests

[the remark this answers is not in the document]
Comment (ravihejmadi) [reply src/SFA.DAS.AANHub.Api/Controllers/AdminsController.cs]: @LiamWright, in the CreateAdmin (Controller + Tests) too please. Just for consistency

**Review by ravihejmadi — APPROVED** [review]

**Pull request merged**